gravitee-node-notifier/src/main/java/io/gravitee/node/notifier/trigger/NotificationTrigger.java

```diff
@@ -0,0 +1,215 @@
+import io.reactivex.Maybe;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.support.CronExpression;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class NotificationTrigger implements Handler<Long> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(
+    NotificationTrigger.class
+  );
+
+  private Vertx vertx;
+
+  private NotificationAcknowledgeRepository notificationAcknowledgeRepository;
+
+  private NotifierPluginFactory notifierFactory;
+
+  private NotificationDefinition definition;
+
+  private NotificationCondition condition;
+
+  private Long scheduledTaskId;
+
+  private CronExpression cronExpression;
+
+  public NotificationTrigger(
+    Vertx vertx,
+    NotificationAcknowledgeRepository notificationAcknowledgeRepository,
+    NotifierPluginFactory notifierFactory
+  ) {
+    this.vertx = vertx;
+    this.notificationAcknowledgeRepository = notificationAcknowledgeRepository;
+    this.notifierFactory = notifierFactory;
+  }
+
+  public void initialize(
+    NotificationDefinition definition,
+    NotificationCondition condition
+  ) {
+    this.definition = definition;
+    this.condition = condition;
+    this.cronExpression = CronExpression.parse(definition.getCron());
+  }
+
+  public void start() {
+    scheduleNextAttempt();
+  }
+
+  private void scheduleNextAttempt() {
+    this.scheduledTaskId = this.vertx.setTimer(computeNextAttempt(), this);
+  }
+
+  public void stop() {
+    if (
+      this.scheduledTaskId != null &&
+      this.vertx.cancelTimer(this.scheduledTaskId)
+    ) {
+      LOGGER.debug("Notification Trigger cancelled !");
+    } else {
+      LOGGER.debug("Notification Trigger can't be cancelled or doesn't exist");
+    }
+  }
+
+  private long computeNextAttempt() {
+    final LocalDateTime now = LocalDateTime.now();
+    return toEpochMillis(this.cronExpression.next(now)) - toEpochMillis(now);
+  }
+
+  private long toEpochMillis(LocalDateTime dateTime) {
+    return dateTime.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();
+  }
+
+  @Override
+  public void handle(Long event) {
+    if (condition.test(definition)) {
+      // keep EventLoop context in order to use it to execute the notifier's send method
+      final Context triggerContext = vertx.getOrCreateContext();
+      this.notificationAcknowledgeRepository.findByResourceIdAndAudienceId(
+          this.definition.getResourceId(),
+          definition.getType(),
+          definition.getAudienceId()
+        )
+        .map(Optional::ofNullable)
+        .switchIfEmpty(Maybe.just(Optional.empty()))
+        .subscribe(
+          notifAck -> {
+            if (notifAck.isEmpty()) {
```

Comment:
Here we never reschedule anything in case of error, is it normal ?
I saw also that, once the notification is successfully sent, no new scheduling is set. I think it makes sense but nothing inform the NotifierService that this trigger is not active anymore and could be removed from the triggers map.